src/store/coinListStore/СoinListStore.ts

```diff
@@ -0,0 +1,100 @@
+import { CoinListModel } from "@store/models";
+import PaginationStore from "@store/paginationStore";
+import rootStore from "@store/RootStore/instance";
+import { Meta } from "@utils/meta";
+import { ILocalStore } from "@utils/useLocalStore";
+import {
+  action,
+  computed,
+  reaction,
+  makeAutoObservable,
+  runInAction,
+} from "mobx";
+
+import DropdownStore from "./DropdownStore";
+import { requestCoinList } from "./requestCoinList";
+import SearchStore from "./SearchStore";
+
+export default class СoinListStore implements ILocalStore {
+  _coins: CoinListModel[] = [];
+  meta: Meta;
+  searchStore;
+  dropdownStore;
+  paginationStore;
+
+  constructor() {
+    this.meta = Meta.initial;
+    this.searchStore = new SearchStore();
+    this.dropdownStore = new DropdownStore();
+    this.paginationStore = new PaginationStore(10);
+    this.paginationStore.page = Number(rootStore.query.getParam("page")) || 1;
+    this.searchStore._search = rootStore.query.getParam("search");
+    this.dropdownStore.dropdownValues.key = rootStore.query.getParam("currency")
+      ? rootStore.query.getParam("currency")
+      : "usd";
+    this.dropdownStore.dropdownValues.value = rootStore.query.getParam(
+      "currency"
+    )
+      ? `Market - ${rootStore.query.getParam("currency").toUpperCase()}`
+      : "Market - USD";
+    makeAutoObservable(this, {
+      fetch: action.bound,
+      coins: computed,
+      serchedCoins: computed,
+    });
+  }
+
+  destroy() {
+    // если диспоузить реакцию она перестает работать
+    // this._queryPaginationReaction();
+  }
+
+  _queryPaginationReaction = reaction(
+    () => this.serchedCoins.length,
+    () => {
+      this.paginationStore.count = this.serchedCoins.length;
+      this.paginationStore.page = 1;
+    }
+  );
```

Comment:
! [в этом и смысл диспоуза реакций](https://mobx.js.org/reactions.html#always-dispose-of-reactions)
предположим у тебя этот стор используется в каком-то компоненте, на анмаунте компонента надо вызывать `destroy` стора, в котором происходит `dispose` реакции, чтобы она вечно не висела в памяти
видимо ты где-то не в том месте вызываешь метод `destroy`
```suggestion
  destroy() {
    this.queryPaginationReactionDisposer();
  }

  queryPaginationReactionDisposer = reaction(
    () => this.serchedCoins.length,
    () => {
      this.paginationStore.count = this.serchedCoins.length;
      this.paginationStore.page = 1;
    }
  );
```